homeassistant/components/bzutech/config_flow.py

```diff
@@ -0,0 +1,203 @@
+    },
+    True,
+)
+
+
+async def validate_input(hass: HomeAssistant, data: dict[str, Any]) -> BzuTech:
+    """Validate the user input allows us to connect."""
+    hass.data[DOMAIN] = {}
+    api = BzuTech(data[CONF_EMAIL], data[CONF_PASSWORD])
+
+    if not await api.start():
+        raise InvalidAuth
+
+    return api
+
+
+def get_devices(api: BzuTech, page: int) -> dict[str, Any]:
+    """Get device names on a dict for the showmenu."""
+
+    devices = {}
+    i = 1
+    first = page * 4
+    last = first + 3
+    counter = 0
+
+    for key in list(api.dispositivos.keys()):
+        if first <= counter <= last:
+            returnkey = "option" + str(i)
+            devices[returnkey] = key
+            i = i + 1
+
+        counter = counter + 1
+    if len(list(api.dispositivos.keys())) > (page + 1) * 4:
+        devices["option5"] = "Mais dispositivos"
+    return devices
+
+
+def get_ports(api: BzuTech, chipid: str) -> dict[str, str]:
+    """Get ports with the endpoints connected to each port."""
+    ports = {}
+    ports["option1"] = "Port 1 " + str(api.get_endpoint_on(chipid, 1))
+    ports["option2"] = "Port 2 " + str(api.get_endpoint_on(chipid, 2))
+    ports["option3"] = "Port 3 " + str(api.get_endpoint_on(chipid, 3))
+    ports["option4"] = "Port 4 " + str(api.get_endpoint_on(chipid, 4))
+
+    return ports
+
+
+class ConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle a config flow for BZUTech."""
+
+    VERSION = 2
+    api: BzuTech
+    email = ""
+    password = ""
+    devicepage = 0
+    flowstep = 0
+    selecteddevice = 0
+    selectedport = 0
+
+    async def async_step_user(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Handle the initial step."""
+        errors: dict[str, str] = {}
+        if user_input is not None:
+            try:
+                self.api = await validate_input(self.hass, user_input)
+            except CannotConnect:
+                errors["base"] = "cannot_connect"
+            except InvalidAuth:
+                errors["base"] = "invalid_auth"
+            except Exception:  # pylint: disable=broad-except
+                _LOGGER.exception("Unexpected exception")
+                errors["base"] = "unknown"
+            else:
+                self.email = user_input[CONF_EMAIL]
+                self.password = user_input[CONF_PASSWORD]
+                return await self.async_step_dispselect(user_input=user_input)
+        return self.async_show_form(
+            step_id="user",
+            data_schema=STEP_USER_LOGIN_SCHEMA,
+            errors=errors,
+            last_step=False,
+        )
+
+    async def async_step_dispselect(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Set up second step."""
+        self.flowstep = 1
+        return self.async_show_menu(
+            step_id="dispselect",
+            menu_options=get_devices(self.api, self.devicepage),
+        )
```

Comment:
I really don't think a menu step is the right choice, based on all the duplicated code below.
Wouldn't `async_show_form` with a select selector work better?